packages/flutter_tools/lib/src/commands/build_aar.dart

```diff
@@ -155,6 +155,8 @@ class BuildAarCommand extends BuildSubCommand {
     if (remainingArguments.isEmpty) {
       return FlutterProject.current();
     }
-    return FlutterProject.fromDirectory(globals.fs.directory(findProjectRoot(globals.fs, remainingArguments.first)));
+    String? projectRoot = findProjectRoot(globals.fs, remainingArguments.first);
+    projectRoot ??= '.';
```

Comment:
Why was this needed? I would think this would be equivalent to passing null and then having `findProjectRoot` fall back to `fileSystem.currentDirectory.path`.

Reply:
oh wait, never mind, this handling a null return from `findProjectRoot`. Although, if `findProjectRoot` returns null, doesn't that mean the current working directory is NOT a valid project?

Reply:
This was to make https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart#L213 pass but after more debugging found out that `build aar` is passing `/path/lib/main.dart' to `findProjectRoot` that with my change when doing the `directory.existsSync` fails and returned a null, the correct behavior is to get the directory in which `main.dart` lives or throw a tool exit if `/path/lib/main.dart` does not exists which currently just falls back to the current directory because of `fileSystem.path.dirname(directory)`